### PR TITLE
Add new operational scenarios

### PR DESCRIPTION
**Pull request by rpappalax**
@bbangert r?

### CODE REVIEW

**Comment by rpappalax** [comment]
oops forgot to fix the tests. i'll fix and resubmit

File: README.md [hunk 1]
```diff
@@ -52,7 +52,7 @@ Run the basic scenario against the dev server:
 Run 5 instances of the basic scenario, starting one every second, against the
 dev server:
 
-    $ aplt_testplan wss://autopush-dev.stage.mozaws.net/ "aplt.scenarios:basic 5 1 0"
+    $ aplt_testplan wss://autopush-dev.stage.mozaws.net/ "aplt.scenarios:basic,5,1,0"
```
Comment (bbangert) [inline README.md]: Good catch!

File: aplt/scenarios.py [hunk 2]
```diff
@@ -39,7 +39,31 @@ def basic():
     yield disconnect()
 
 
-def basic_forever(notif_delay=300, run_once=0):
+def register_forever(regist_delay=0, run_once=0):
```
Comment (bbangert) [inline aplt/scenarios.py]: Might want a default delay.

**Comment by rpappalax** [comment]
@bbangert I have a few new scenarios to add.  should i add them to this PR or a separate one?

**Comment by rpappalax** [comment]
Closes #44

File: aplt/scenarios.py [hunk 3]
```diff
@@ -57,30 +131,66 @@ def basic_forever(notif_delay=300, run_once=0):
...
         yield wait(notif_delay)
+
         if run_once:
+            yield unregister(reg["channelID"])
             yield disconnect()
             break
 
 
-def connect_and_idle_forever():
+def notification_forever_stored(qty_stored=1000, ttl=300, flood_delay=30,
```
Comment (rpappalax) [inline aplt/scenarios.py]: I'll prob need to tweak these delays if we want stored notifications to be delivered before TTL is over -or- if we want them to be delivered in a flood (i.e. flood_delay=0 for this scenario)

**Comment by bbangert** [comment]
The commits need to be squashed into a single commit that indicates what this does.

File: aplt/scenarios.py [hunk 4]
```diff
@@ -57,30 +131,66 @@ def basic_forever(notif_delay=300, run_once=0):
...
+                                notif_delay=30, run_once=0):
+    """Connects, then repeats every delay interval:
+    1. register
+    2. send notifications x qty_stored (# of notifications to store)
+    3. wait for flood_delay (seconds)
+    4. receive notifications x qty_stored
+
+    Repeats forever.
+    """
```
Comment (bbangert) [inline aplt/scenarios.py]: This won't consistently work right now, until #41 is addressed. Upcoming changes may cause messages to be delivered slightly out of order, so relying on the order as this does is a bit brittle.

File: aplt/utils.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+def string_generator(length=10):
+    import string
+    import random
+    return ''.join(random.choice(string.ascii_lowercase + string.digits)
+                   for _ in range(length))
+
+
+def bad_push_endpoint(push_endpoint=None, token_length=None):
+    from random import randint
```
Comment (bbangert) [inline aplt/utils.py]: Imports go under the module docs.

File: aplt/utils.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+    """Given a valid endpoint URL, return a new one with an invalid
+    token of token_length.
```
Comment (bbangert) [inline aplt/utils.py]: Add line-break after the period.

File: aplt/utils.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    If token_length not specified, append a random token string of
+    random length.
+    If no push_endpoint specified, return a bogus endpoint with an
+    invalid token."""
```
Comment (bbangert) [inline aplt/utils.py]: Line-break before closing triple-quotes.

**Comment by bbangert** [comment]
Overall this looks good, just a few minor changes, and probably removing that bit that relies on message ordering until #44 is fixed.

**Comment by rpappalax** [comment]
@bbangert all changes in and squashed down.  is it good to go?

**Comment by bbangert** [comment]
The code looks good, I would change the commit message slightly. I can imagine many 'add new loadtest scenarios' commit messages, so it'd be nice if the commit message was distinct about the nature of these so that it'd be meaningful when browsing commit history to know when a particular set was added.

**Comment by rpappalax** [comment]
@bbangert does that work?

**Comment by bbangert** [comment]
Works for me.

**Comment by bbangert** [comment]
r+

**Pull request merged**